src/Appwrite/Resque/Worker.php

```diff
@@ -177,56 +179,84 @@ protected function getProjectDB(string $projectId): Database
         return $this->getDB(self::DATABASE_PROJECT, $projectId, $project->getInternalId());
     }
 
+    /**
+     * Get internal project database given the project document
+     *
+     * Allows avoiding race conditions when modifying the projects collection
+     * @param Document $project
+     * @return Database
+     * @throws Exception
+     */
+    protected function getProjectDBFromDocument(Document $project): Database
```

Comment:
Hmm is this going to cause a lot of conflict with:

https://github.com/appwrite/appwrite/blob/708994d4a66838ece5ae0b37a2902bf6bda16e2e/src/Appwrite/Resque/Worker.php#L175